pkg/parse/updater.go

```diff
@@ -69,17 +72,22 @@ func (u *updater) Errors() status.MultiError {
 }
 
 func (u *updater) setValidationErrs(errs status.MultiError) {
-	u.mux.Lock()
-	defer u.mux.Unlock()
+	u.errorMux.Lock()
+	defer u.errorMux.Unlock()
 	u.validationErrs = errs
 }
 
 func (u *updater) setWatchErrs(errs status.MultiError) {
-	u.mux.Lock()
-	defer u.mux.Unlock()
+	u.errorMux.Lock()
+	defer u.errorMux.Unlock()
 	u.watchErrs = errs
 }
 
+// Updating returns true if the Update method is running.
+func (u *updater) Updating() bool {
```

Comment:
This smells a bit like the other bool. Is the idea that by setting this at a higher level (rather than reaching into the applier to get the value) it is less prone to race conditions?

Reply:
Moving it up a level allows including the other actions performed by `updater.Update()`.

Setting the bool inside the newly added mutex ensures that the bool value is not affected by race conditions.

But yes, it's a lazy fix, not an ideal fix. Ideally the control flow would be modified to use channels, but I decided to start with the quick fix, because I've had trouble getting large rewrites tested and approved. And I'd like to get back to finishing the finalizer.

Reply:
The errors and syncing bool could be bundled together in a Status() result or something, to be atomic, and the current status could be the result of a for loop that reads from a channel accepting errors and start/stop signals.

But once you go down that road, you realize that the applier, updater, parser, and reconciler layers ALL have the same problematic pattern and ALL need to be rewritten to use a top level reconcile loop. Then you wouldn't need to synchronize any of these lower layers.